pkg/domain/infra/abi/system.go

```diff
@@ -179,7 +180,16 @@ func (ic *ContainerEngine) SystemPrune(ctx context.Context, options entities.Sys
 			found = true
 		}
 		systemPruneReport.PodPruneReport = append(systemPruneReport.PodPruneReport, podPruneReport...)
-		containerPruneReport, err := ic.pruneContainersHelper(nil)
+		containerPruneOptions := entities.ContainerPruneOptions{}
+		for _, f := range options.Filter {
+			t := strings.SplitN(f, "=", 2)
+			containerPruneOptions.Filters = make(url.Values)
+			if len(t) < 2 {
+				return nil, errors.Errorf("filter input must be in the form of filter=value: %s is invalid", f)
+			}
+			containerPruneOptions.Filters.Add(t[0], t[1])
+		}
```

Comment:
When I was messing with this feature I noticed that this pattern is repeated in some form or another

```
cmd/podman/containers/prune.go:69:		t := strings.SplitN(f, "=", 2)
cmd/podman/containers/ps.go:189:		split := strings.SplitN(f, "=", 2)
cmd/podman/networks/list.go:70:		split := strings.SplitN(f, "=", 2)
cmd/podman/pods/ps.go:85:			split := strings.SplitN(f, "=", 2)
cmd/podman/volumes/list.go:76:		filterSplit := strings.SplitN(f, "=", 2)
libpod/image/prune.go:116:		filterSplit := strings.SplitN(f, "=", 2)
libpod/image/search.go:266:		arr := strings.SplitN(f, "=", 2)
```

Could this be abstracted into a helper function or is it not worth it? I thought maybe putting a `parseFilterArgumentsIntoMap` function into `libpod/util.go` or `cmd/podman/utils/utils.go` but I wasn't sure.

Reply:
SGTM.  I think getting them altogether would be good.